openest/generate/functions.py

```diff
@@ -443,6 +443,148 @@ def describe():
                     description="Sum the results of multiple previous calculations.")
 
 
+
+class FractionSum(calculation.Calculation):
+    """Sum of two subcalculations weighted on fraction of unity.
+
+    In short, ``frac * subcalc[0] + (1 - frac) * subcalc[1]``.
+
+    Parameters
+    ----------
+    frac : ``openest.generate.calculation.Calculation``
+        Result gives weight of first subcalculation. 1 - `frac` is weight of
+        second subcalculation. Must be ≤ 1. The
+    subcalcs : Sequence of ``openest.generate.calculation.Calculation``
+        Calculations to weight. Must be ``len(subcalcs) == 2``. The units of
+        the subcalcs elements must be the same.
+    unshift : bool, optional
+    """
+
+    def __init__(self, frac, subcalcs, unshift=True):
+        if len(subcalcs) != 2:
+            raise ValueError('Length of `subcalcs` must be 2')
+
+        fullunitses = subcalcs[0].unitses[:]
+        for ii in range(1, len(subcalcs)):
+            assert subcalcs[0].unitses[0] == subcalcs[ii].unitses[0], "%s <> %s" % (subcalcs[0].unitses[0], subcalcs[ii].unitses[0])
+            fullunitses.extend(subcalcs[ii].unitses)
+        if unshift:
+            super().__init__([subcalcs[0].unitses[0]] + fullunitses)
+        else:
+            super().__init__([subcalcs[0].unitses[0]])
+
+        self.unshift = unshift
+        self.subcalcs = subcalcs
+        self.frac = frac
+
+
+    def format(self, lang, *args, **kwargs):
+        raise NotImplementedError
+        mains = []
+        elements = {}
+        alldeps = set()
+        for subcalc in self.subcalcs:
+            elements.update(subcalc.format(lang, *args, **kwargs))
+            mains.append(elements['main'])
+            alldeps.update(elements['main'].dependencies)
+
+        if lang in ['latex', 'julia']:
+            elements['main'] = FormatElement(' * '.join([main.repstr for main in mains]), list(alldeps))
```

Comment:
Fix this.

Reply:
We're actually raising a `NotImplementedError` before we get to this.

I'm not sure how to handle this for the Calculation.

Reply:
Resolved with 26b0a1c.